lib/ansible/modules/cloud/azure/azure_rm_cdnendpoint.py

```diff
@@ -0,0 +1,677 @@
+                    self.log("Can't stop/stop a non-existed endpoint")
+                    self.fail("This endpoint is not found, stop/start is forbidden")
+
+            else:
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(response['tags'])
+
+                if response['provisioning_state'] == "Succeeded":
+
+                    if self.started is False and response['resource_state'] == 'Running':
+                        self.log("Need to stop the Azure CDN endpoint")
+
+                        if not self.check_mode:
+                            result = self.stop_cdnendpoint()
+                            self.results['id'] = result['id']
+                            self.results['host_name'] = result['host_name']
+                            self.log("Endpoint stopped")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    elif self.started and response['resource_state'] == 'Stopped':
+                        self.log("Need to start the Azure CDN endpoint")
+
+                        if not self.check_mode:
+                            result = self.start_cdnendpoint()
+                            self.results['id'] = result['id']
+                            self.results['host_name'] = result['host_name']
+                            self.log("Endpoint started")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    elif self.started is not None:
+                        self.log("Start/Stop not performed due to current resource state")
+                        self.results['id'] = response['id']
+                        self.results['host_name'] = response['host_name']
+                        self.results['changed'] = False
+                        return self.results
+
+                    if self.purge:
+                        self.log("Need to purge endpoint")
+
+                        if not self.check_mode:
+                            result = self.purge_cdnendpoint()
+                            self.results['id'] = result['id']
+                            self.results['host_name'] = result['host_name']
+                            self.log("Endpoint purged")
+
+                        self.results['changed'] = True
+                        return self.results
+
+                    if update_tags:
+                        to_be_updated = True
+
+                    to_be_updated = to_be_updated or self.check_update(response)
+
+                    if to_be_updated:
+                        self.log("Need to update the Azure CDN endpoint")
+
+                        if not self.check_mode:
+                            result = self.update_cdnendpoint()
+                            self.results['id'] = result['id']
+                            self.results['host_name'] = result['host_name']
```

Comment:
This one *might* not be redundant, if it's possible to make changes that will cause the CDN endpoint hostname to change...

Reply:
fixed